express/parsers/apps/espresso/parser.py

```diff
@@ -51,14 +64,34 @@ def nspins(self):
         """
         return self.xml_parser.nspins()
 
+    def _is_gw_calculation(self):
+        """
+        Checks whether this is a GW calculation.
+
+        NOTE: DO NOT READ THE WHOLE FILE INTO MEMORY.
+
+        Returns:
+             bool
+        """
+        with open(self.stdout_file, "r") as f:
+            for index, line in enumerate(f):
+                if index > 50: return False
+                if "SternheimerGW" in line:
+                    return True
```

Comment:
We need to explain the logic in docstring - what do you actually do/assume here about the file?